subkey/subhd/src/lib.rs

```diff
@@ -70,10 +74,16 @@ pub struct Wookong;
 
 impl Wallet for Wookong {
 	type Pair = sr25519::Pair;
-	fn derive_public(&self, _path: &[DeriveJunction]) -> Result<Public> {
-		// TODO: pass through _path and _password to mutate the key on the device accordingly.
+	fn derive_public(&self, path: &[DeriveJunction]) -> Result<Public> {
 		let mut pk: [u8; 32] = [0u8; 32];
-		let rv = wk_getpub(&mut pk);
+		let mut vec = Vec::new();
+		for j in path {
```

Comment:
I am no expert in derivation paths nor the hardware wallet integration, but regardless I don't get this loop: 

you are matching to check the type of junction, but you are doing the same thing regardless of the result?

Reply:
indeed; this is definitely wrong behaviour. the device needs to derive correctly according to the path.